window/src/os/windows/window.rs

```diff
@@ -1579,14 +1579,20 @@ unsafe fn key(hwnd: HWND, msg: UINT, wparam: WPARAM, lparam: LPARAM) -> Option<L
             // We set RIGHT_ALT as a hint to ourselves that AltGr is in
             // use (we use regular ALT otherwise) so that our dead key
             // resolution can do the right thing.
-            //
+            modifiers |= Modifiers::RIGHT_ALT;
+        } else if config().treat_left_ctrlalt_as_altgr()
```

Comment:
Should also constrain this to layouts with AltGr mappings:

```suggestion
        } else if inner.keyboard_info.has_alt_gr()
 && config().treat_left_ctrlalt_as_altgr()
```

Reply:
Done